src/Compilers/Core/Desktop/AnalyzerFileReference.InMemoryAssemblyLoader.cs

```diff
@@ -247,6 +247,10 @@ private static Assembly ResolveForKnownRequestor(string requestedAssemblyName, A
 
                     string directoryPath = Path.GetDirectoryName(requestingAssemblyFullPath);
                     string assemblyFullPath = Path.Combine(directoryPath, assemblyIdentity.Name + ".dll");
+                    if (!File.Exists(assemblyFullPath))
+                    {
+                        return null;
+                    }
 
                     assembly = LoadCore(assemblyFullPath);
```

Comment:
A safer fix would be to wrap `LoadCore()` in try/catch to cover other scenarios (however unlikely), e.g., the file exists but cannot be opened or the call to `Assembly.Load()` fails.